src/components/PiecesPreviewModal/PiecesPreviewModalForm.js

```diff
@@ -41,6 +45,34 @@ const PiecesPreviewModalForm = ({
   const { change } = useForm();
   const intl = useIntl();
 
+  useEffect(() => {
+    change('numberOfCycles', 1);
+  }, [change]);
+
+  const TIME_UNIT_PER_YEAR = {
+    day: { timeUnitsPerYear: 365.2425 },
+    week: { timeUnitsPerYear: 52.1775 },
+    month: { timeUnitsPerYear: 12 },
+  };
+
+  const MAX_PIECES = 366;
+
+  const calculateAllowedCycles = () => {
+    const { issues, period, timeUnit } = ruleset?.recurrence ?? {};
+
+    if (timeUnit?.value === 'year') {
+      return Math.floor((MAX_PIECES / issues));
+    } else {
+      const timeUnitsPerYear = TIME_UNIT_PER_YEAR[timeUnit?.value]?.timeUnitsPerYear;
+      return Math.floor(MAX_PIECES / ((timeUnitsPerYear / period) * issues));
+    }
+  };
```

Comment:
Implementation here looks good, could do with some tidying up, can be wrapped in `useMemo` for performance so that its only calculated once on a ruleset change, additionally wont need to be recalculate when called in custom validation method:

![image](https://github.com/user-attachments/assets/1dc9b94d-4c45-42fa-befd-e23096b7fa56)

Additionally can remove method calls from places where this is used meaning:
`values={{ maxValue: calculateAllowedCycles() }}` becomes `values={{ maxValue: calculateAllowedCycles }}`

Reply:
This can be ignored my changes have been pushed to this branch